### PR TITLE
Metric chart footers

### PR DESCRIPTION
**Pull request by chasulin**
- Removes summary text and info tooltip from above chart
- Implements new footer format with info modal

### CODE REVIEW

File: src/components/Dialogs/utils.ts [hunk 1]
```diff
@@ -8,8 +8,17 @@ export interface MetricModalContent {
...
+  learnLink: string;
 }
 
+const metricToLearnLink: { [key in Metric]: string } = {
+  [Metric.CASE_DENSITY]: '/covid-risk-levels-metrics#daily-new-cases',
+  [Metric.CASE_GROWTH_RATE]: '/covid-risk-levels-metrics#infection-rate',
+  [Metric.HOSPITAL_USAGE]: '/covid-risk-levels-metrics#icu-capacity-used',
+  [Metric.POSITIVE_TESTS]: '/covid-risk-levels-metrics#positive-test-rate',
+  [Metric.VACCINATIONS]: '/covid-risk-levels-metrics#percent-vaccinated',
+};
```
Comment (pnavarrc) [inline src/components/Dialogs/utils.ts:20]: Can we grab this from the CMS? `src/cms-content/tooltips/metric-calculation-tooltips.json` has the CTA (even if we are not reusing the title)

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 2]
```diff
@@ -28,56 +24,106 @@ export interface ShareButtonProps {
   showEmbedButton?: boolean;
 }
 
+interface DialogProps {
+  open: boolean;
+  closeDialog: any;
+  openDialog: any;
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:30]: Maybe give these the correct types?

File: src/components/Dialogs/utils.ts (hunk 1 above)
Comment (chasulin) [reply src/components/Dialogs/utils.ts:20]: and by down the line i mean.. tomorrow or thursday

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx (hunk 2 above)
Comment (chasulin) [reply src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:30]: oh whoops! yep

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
   AboutText,
   ModalButton,
-  DisclaimerWrapper,
+  Wrapper,
+  OverrideDisclaimer,
 } from './ChartFooter.style';
-import { DialogMain } from 'components/Dialogs';
-import { Link } from 'cms-content/modals';
+import { DialogMain, MetricInfoDialogInner } from 'components/Dialogs';
+// import { Link } from 'cms-content/modals';
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:12]: we can probably remove that

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 4]
```diff
@@ -28,56 +24,106 @@ export interface ShareButtonProps {
...
+  open: boolean;
+  closeDialog: () => void;
+  openDialog: () => void;
+  modalContent: MetricModalContent;
+  modalHeader: string;
+}
+
+const ShareButtonBlock: React.FC<ShareButtonProps> = ({
+  ...shareButtonProps
+}) => {
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:37]: Isn't this the same than 

```tsx
const ShareButtonBlock: React.FC<ShareButtonProps> = (shareButtonProps) => {
  // ...
};
```

Comment (chasulin) [reply src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:37]: whoops, yes

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 5]
```diff
@@ -28,56 +24,106 @@ export interface ShareButtonProps {
...
+  const dialogProps = {
+    open,
+    closeDialog,
+    header: modalHeader,
+    links: [
+      {
+        cta: 'Learn more',
+        url: modalContent.learnLink,
+      },
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:60]: We should add a good `aria-label` for this link, although given that the modal covers a single metric, a screen reader user will probably know that the link is to Lean more about that metric.

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 6]
```diff
@@ -28,56 +24,106 @@ export interface ShareButtonProps {
...
+
+const ShareButtonBlock: React.FC<ShareButtonProps> = ({
+  ...shareButtonProps
+}) => {
+  return (
+    <ButtonContainer>
+      <ShareButtons {...shareButtonProps} />
+    </ButtonContainer>
+  );
+};
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:43]: Not sure if we need this component, is really small 😃

Comment (chasulin) [reply src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:43]: yeah, its just repeated twice down in the main component, so i pulled it out into its own thing

File: public/admin/config.yml [hunk 7]
```diff
@@ -899,3 +899,7 @@ collections:
             label: URL
             widget: string
             hint: 'Ex: /covid-risk-levels-metrics#vulnerability'
+          - name: ariaLabel
```
Comment (pnavarrc) [inline public/admin/config.yml:902]: 🙏

File: src/components/LocationPage/ChartBlock.tsx [hunk 8]
```diff
@@ -40,55 +35,50 @@ function ChartBlock(props: {
...
-              showEmbedButton={false}
-            />
-          </LocationPageSectionFooter>
+          <MetricChartFooter
+            footerText={getMetricStatusText(metric, projections)}
+            overrideDisclaimer={overrideDisclaimer}
+            shareButtonProps={shareButtonProps}
+            modalContent={modalContent}
+            modalHeader={getMetricName(metric)}
+          />
```
Comment (pnavarrc) [inline src/components/LocationPage/ChartBlock.tsx:70]: How about passing `region`, `projection` and `metric` as props to `MetricChartFooter` and calculating each part in that component instead? That would make the API of `MetricChartFooter` look a bit more like `MetricChart` and hide implementation details that are not relevant in `ChartBlock`

File: src/components/LocationPage/ChartBlock.tsx [hunk 9]
```diff
@@ -40,55 +35,50 @@ function ChartBlock(props: {
...
         </>
       )}
     </>
   );
 }
 
-export function renderDisclaimer(
+export function getOverrideDisclaimer(
```
Comment (pnavarrc) [inline src/components/LocationPage/ChartBlock.tsx:77]: This function could be moved to `MetricChartFooter` too

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 10]
```diff
@@ -1,25 +1,20 @@
...
 import type { MetricValues } from 'common/models/Projections';
-import { MarkdownBody } from 'components/Markdown';
-
-export interface AboutModalProps {
-  header: string;
-  body: string;
-  links: Link[];
-}
+import { useDialog } from 'common/hooks';
+import { MetricModalContent } from 'components/Dialogs/utils';
```
Comment (pnavarrc) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:17]: Can this import be merged with the other import from `component/Dialogs`?

File: src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx [hunk 11]
```diff
@@ -28,56 +31,115 @@ export interface ShareButtonProps {
...
+  region: Region;
+  stats: any;
+}> = ({ metric, projections, region, stats }) => {
+  const provenance = getSourcesForMetric(
+    projections.primary.annotations,
+    metric,
+  );
+  const overrideDisclaimer = getOverrideDisclaimer(region, metric, provenance);
+  const modalContent = getMetricModalContent(region, metric, provenance);
+  const modalHeader = getMetricNameExtended(metric);
```
Comment (chasulin) [inline src/components/NewLocationPage/ChartFooter/MetricChartFooter.tsx:94]: using extended metric name so the percent vaccinated modal header isnt just 'Vaccinated' (this change also changes Daily new cases --> Daily new cases per 100k population, which doesnt seem like an issue)

**Review by pnavarrc — APPROVED** [review]
Looks good! 🚢

**Pull request merged**